### PR TITLE
add option to set/unset user-agent header

### PR DESCRIPTION
**Pull request by resdenia**
To solve the `cors` issue in firefox we need to  give the possibility to send logs without `user-agent` header.

Group 2293

### CODE REVIEW

File: lib/logzio-nodejs.js [hunk 1]
```diff
@@ -53,6 +53,7 @@ class LogzioLogger {
         timeout,
         sleepUntilNextRetry = 2 * 1000,
         callback = this._defaultCallback,
+		setUserAgent = true,
```
Comment (yotamloe) [inline lib/logzio-nodejs.js]: Please fix the indentation :)

File: lib/logzio-nodejs.js [hunk 2]
```diff
@@ -71,7 +72,7 @@ class LogzioLogger {
         this.compress = compress;
         this.internalLogger = internalLogger;
         this.sleepUntilNextRetry = sleepUntilNextRetry;
-
+		this.setUserAgent = setUserAgent;
```
Comment (yotamloe) [inline lib/logzio-nodejs.js]: Please fix the indentation :)

**Review by yotamloe — CHANGES_REQUESTED** [review]
Overall looks good, Please add a description to this PR and a unit test for the new feature.

File: test/logger.test.js [hunk 3]
```diff
@@ -59,6 +59,21 @@ describe('logger', () => {
...
+            const logger = createLogger({
+                bufferSize: 1,
+                callback: done,
+				setUserAgent:false
+            });
+            sinon.spy(logger, '_createBulk');
+
+            const logMsg = 'hello there from test';
+            logger.log(logMsg);
+            assert.equal(logger._createBulk.getCall(0).args[0][0].message, logMsg);
```
Comment (yotamloe) [inline test/logger.test.js]: This `assert` is validating the log message and not the `user-agent` header

**Review by yotamloe — APPROVED** [review]
LGTM

File: README.md [hunk 4]
```diff
@@ -47,6 +47,7 @@ logger.log(obj);
 * **addTimestampWithNanoSecs** - Add a timestamp with nano seconds granularity. This is needed when many logs are sent in the same millisecond, so you can properly order the logs in kibana. The added timestamp field will be `@timestamp_nano` Default: `false`
 * **compress** - If true the the logs are compressed in gzip format. Default: `false`
 * **internalLogger** - set internal logger that supports the function log. Default: console.
+* **setUserAgent** - To sending logs we are providing for user to send logs with user-agent. To send logs without user-agent please change to `false`.  Default:`true`,
```
Comment (Doron-Bargo) [inline README.md]: This is not clear

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
Please update readme

File: lib/logzio-nodejs.js [hunk 5]
```diff
@@ -253,14 +254,24 @@ class LogzioLogger {
...
+        let headersOpt;
+        if (this.setUserAgent) {
+            headersOpt = {
                 host: this.host,
                 accept: '*/*',
                 'user-agent': USER_AGENT,
                 'content-type': 'text/plain',
-            },
+            };
+        } else {
```
Comment (AlonMiz) [inline lib/logzio-nodejs.js]: Y else just add the user agent to the object?

File: lib/logzio-nodejs.js [hunk 6]
```diff
@@ -253,14 +254,24 @@ class LogzioLogger {
...
-        const options = {
-            uri: this.url,
-            headers: {
+        let headersOpt;
+        if (this.setUserAgent) {
+            headersOpt = {
                 host: this.host,
                 accept: '*/*',
                 'user-agent': USER_AGENT,
                 'content-type': 'text/plain',
```
Comment (AlonMiz) [inline lib/logzio-nodejs.js]: Eg.
```suggestion
                'content-type': 'text/plain',
                ...(userAgent? {'user-agent': USER_AGENT} : {})
```

Comment (resdenia) [reply lib/logzio-nodejs.js]: Yes!
Thank you @AlonMiz !

**Review by Doron-Bargo — APPROVED** [review]
lgtm

**Review by Doron-Bargo — APPROVED** [review]
lgtm

**Review by Doron-Bargo — APPROVED** [review]
lgtm

**Pull request merged**